modules/iam/main.tf

```diff
@@ -182,84 +211,145 @@ EOF
+      "ec2:AuthorizeSecurityGroupIngress",
+      "ec2:CancelSpotInstanceRequests",
+      "ec2:CreateNetworkInterface",
+      "ec2:CreateSecurityGroup",
+      "ec2:CreateTags",
+      "ec2:DeleteNetworkInterface",
+      "ec2:DeleteSecurityGroup",
+      "ec2:DeleteTags",
+      "ec2:DescribeAvailabilityZones",
+      "ec2:DescribeAccountAttributes",
+      "ec2:DescribeDhcpOptions",
+      "ec2:DescribeImages",
+      "ec2:DescribeInstanceStatus",
+      "ec2:DescribeInstances",
+      "ec2:DescribeKeyPairs",
+      "ec2:DescribeNetworkAcls",
+      "ec2:DescribeNetworkInterfaces",
+      "ec2:DescribePrefixLists",
+      "ec2:DescribeRouteTables",
+      "ec2:DescribeSecurityGroups",
+      "ec2:DescribeSpotInstanceRequests",
+      "ec2:DescribeSpotPriceHistory",
+      "ec2:DescribeSubnets",
+      "ec2:DescribeTags",
+      "ec2:DescribeVpcAttribute",
+      "ec2:DescribeVpcEndpoints",
+      "ec2:DescribeVpcEndpointServices",
+      "ec2:DescribeVpcs",
+      "ec2:DetachNetworkInterface",
+      "ec2:ModifyImageAttribute",
+      "ec2:ModifyInstanceAttribute",
+      "ec2:RequestSpotInstances",
+      "ec2:RevokeSecurityGroupEgress",
+      "ec2:RunInstances",
+      //      "ec2:TerminateInstances",
+      //      "ec2:DeleteVolume",
+      "ec2:DescribeVolumeStatus",
+      "ec2:DescribeVolumes",
+      //      "ec2:DetachVolume",
+      "iam:GetRole",
+      "iam:GetRolePolicy",
+      "iam:ListInstanceProfiles",
+      "iam:ListRolePolicies",
+      "iam:PassRole",
+      "s3:CreateBucket",
+      //      "s3:Get*",
+      //      "s3:List*",
+      "sdb:BatchPutAttributes",
+      "sdb:Select",
+      //      "sqs:CreateQueue",
+      //      "sqs:Delete*",
+      //      "sqs:GetQueue*",
+      //      "sqs:PurgeQueue",
+      //      "sqs:ReceiveMessage",
+      "cloudwatch:PutMetricAlarm",
+      "cloudwatch:DescribeAlarms",
+      "cloudwatch:DeleteAlarms",
+      "application-autoscaling:RegisterScalableTarget",
+      "application-autoscaling:DeregisterScalableTarget",
+      "application-autoscaling:PutScalingPolicy",
+      "application-autoscaling:DeleteScalingPolicy",
+      "application-autoscaling:Describe*"
+    ]
+    resources = [
+      "*",
+    ]
+  }
+
+  statement {
+    actions   = [
+      "ec2:DeleteVolume",
+      "ec2:TerminateInstances",
+      "ec2:DetachVolume",
+    ]
+    resources = [
+      "*"
+    ]
+    # Check that were created by Segment
+    //    condition {
+    //      test     = ""
+    //      values   = []
+    //      variable = ""
+    //    }
+    effect    = "Allow"
+  }
+
+  statement {
+    actions   = [
+      "sqs:CreateQueue",
+      "sqs:Delete*",
+      "sqs:GetQueue*",
+      "sqs:PurgeQueue",
+      "sqs:ReceiveMessage",
+    ]
+    resources = [
+      "*"
+    ]
+    # Check that were created by Segment
+    //    condition {
+    //      test     = ""
+    //      values   = []
+    //      variable = ""
+    //    }
+    effect    = "Allow"
+  }
+
+  statement {
```

Comment:
I think this one is too broad, as this would allow Segment to access all the S3 objects not explicitly restricted in their `BucketPolicy`